File: Python/MATH/1214 Acima da Média.py
# SUBPROGRAMA
def calcularMedia(notasDosAlunos):
	somaNotas = 0
	for i in range(len(notasDosAlunos)):
		somaNotas += notasDosAlunos[i]
	mediaDasNotas = somaNotas/len(notasDosAlunos)
	return mediaDasNotas


def acimaDaMedia(notasDosAlunos, mediaDasNotas):
	contador = 0
	for i in range(len(notasDosAlunos)):
		if notasDosAlunos[i] > mediaDasNotas:
			contador += 1
	print("%.3f%%" % ((100*contador)/len(notasDosAlunos)))
	return None


# PROGRAMA PRINCIPAL

turmas = int(input())
while turmas != 0:
	notas = list(map(int , input().split()[1:]))
	media = calcularMedia(notas)
	acimaDaMedia(notas, media)
	turmas -= 1

File: C/MATHEMATICS/1214 Acima da Média.c
#include<stdio.h>
#include <stdlib.h>

float calcularMedia(float *notasDosAlunos, float qtdAlunos);
int alunosAcimaDaMedia(float *notasDosAlunos, float qtdAlunos, float mediaDasNotas);

int main()
{
    int turmas; //Quantidade dd casos de testes
    scanf("%d", &turmas);
    while (turmas != 0)
    {
        float *notas, alunos;
        scanf("%f", &alunos);
        notas = (float*) malloc(alunos*sizeof(float));
        for(int i=0; i < alunos; i++)
            scanf("%f", &notas[i]);
        float media;    
        media = calcularMedia(notas, alunos);
        alunosAcimaDaMedia(notas, alunos, media);
        free(notas);
        turmas -= 1;
    }
    return 0;
}

float calcularMedia(float *notasDosAlunos, float qtdAlunos)
{
    float somaDasNotas=0;
    for (int i=0; i < qtdAlunos; i++)
        somaDasNotas += notasDosAlunos[i];
    return (somaDasNotas/qtdAlunos);    
}

int alunosAcimaDaMedia(float *notasDosAlunos, float qtdAlunos, float mediaDasNotas)
{
    float contador=0; // contar alunos acima da media
    for (int i=0; i < qtdAlunos; i++)
    {
        if (notasDosAlunos[i] > mediaDasNotas)
            contador += 1;
    }
    printf("%.3f%%\n", ((100.0*contador)/qtdAlunos));
    return 0;
}
